File: 은빈/파이썬/[PGS]캐릭터의좌표.py
def solution(keyinput, board):
    answer = [0, 0]
    x = (board[0] - 1) // 2
    y = (board[1] - 1) // 2
    
    for d in keyinput:
        if d == "up":
            if answer[1] < y:
                answer[1] += 1
        elif d == "down":
            if answer[1] > -y:
                answer[1] -= 1
        elif d == "left":
            if answer[0] > -x:
                answer[0] -= 1
        elif d == "right":
            if answer[0] < x:
                answer[0] += 1
    
    return answer

File: 은빈/자바/[PGS]캐릭터의좌표.java
class Solution {
    public int[] solution(String[] keyinput, int[] board) {
        int[] answer = new int[2];
        int x = (board[0] - 1) / 2;
        int y = (board[1] - 1) / 2;
        
        for(String d : keyinput)  {
            switch (d) {
                case "up":
                    if(answer[1] < y) {
                        answer[1]++;
                    }
                    break;
                case "down":
                    if(answer[1] > -y) {
                        answer[1]--;
                    }
                    break;
                case "left":
                    if(answer[0] > -x) {
                        answer[0]--;
                    }
                    break;
                case "right":
                    if(answer[0] < x)  {
                        answer[0]++;
                    }
                    break;
            }
        }
        
        return answer;
    }
}